src/net.cpp

```diff
@@ -2063,20 +2063,15 @@ class CNetCleanup
 instance_of_cnetcleanup;
 
 
-
-
-
-
-
-void RelayTransaction(const CTransaction& tx)
+void RelayTransaction(const CTransaction& tx, CAmount fee)
```

Comment:
`CAmount fee` implies an absolute fee. Maybe `CAmount feeRate`?

Personally, I'd prefer to use `CFeeRate` in places except `CNode`. I.e. something like this should suffice:

```diff
diff --git a/src/main.cpp b/src/main.cpp
index 487564a..c0fccbf 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -944 +944 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
-                              bool* pfMissingInputs, CAmount* txFeeRate, bool fOverrideMempoolLimit, const CAmount nAbsurdFee,
+                              bool* pfMissingInputs, CFeeRate* txFeeRate, bool fOverrideMempoolLimit, const CAmount nAbsurdFee,
@@ -1101 +1101 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
-            *txFeeRate = CFeeRate(nFees, nSize).GetFeePerK();
+            *txFeeRate = CFeeRate(nFees, nSize);
@@ -1362 +1362 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
-                        bool* pfMissingInputs, CAmount* txFeeRate, bool fOverrideMempoolLimit, const CAmount nAbsurdFee)
+                        bool* pfMissingInputs, CFeeRate* txFeeRate, bool fOverrideMempoolLimit, const CAmount nAbsurdFee)
@@ -4829 +4829 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
-        CAmount txFeeRate = 0;
+        CFeeRate txFeeRate = CFeeRate(0);
@@ -4863 +4863 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
-                    CAmount orphanFeeRate = 0;
+                    CFeeRate orphanFeeRate = CFeeRate(0);
diff --git a/src/main.h b/src/main.h
index 5c006b2..fe4abb5 100644
--- a/src/main.h
+++ b/src/main.h
@@ -293 +293 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
-                        bool* pfMissingInputs, CAmount* txFee, bool fOverrideMempoolLimit=false, const CAmount nAbsurdFee=0);
+                        bool* pfMissingInputs, CFeeRate* txFee, bool fOverrideMempoolLimit=false, const CAmount nAbsurdFee=0);
diff --git a/src/net.cpp b/src/net.cpp
index 294d2d4..22e8b26 100644
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -2066 +2066 @@ instance_of_cnetcleanup;
-void RelayTransaction(const CTransaction& tx, CAmount fee)
+void RelayTransaction(const CTransaction& tx, CFeeRate fee)
@@ -2074 +2074 @@ void RelayTransaction(const CTransaction& tx, CAmount fee)
-void RelayTransaction(const CTransaction& tx, CAmount fee, const CDataStream& ss)
+void RelayTransaction(const CTransaction& tx, CFeeRate fee, const CDataStream& ss)
@@ -2097 +2097 @@ void RelayTransaction(const CTransaction& tx, CAmount fee, const CDataStream& ss
-            if (fee < pnode->minFeeFilter)
+            if (fee.GetFeePerK() < pnode->minFeeFilter)
diff --git a/src/net.h b/src/net.h
index 0b3400b..3d4117d 100644
--- a/src/net.h
+++ b/src/net.h
@@ -776,2 +776,2 @@ class CTransaction;
-void RelayTransaction(const CTransaction& tx, CAmount fee);
-void RelayTransaction(const CTransaction& tx, CAmount fee, const CDataStream& ss);
+void RelayTransaction(const CTransaction& tx, CFeeRate fee);
+void RelayTransaction(const CTransaction& tx, CFeeRate fee, const CDataStream& ss);
diff --git a/src/rpc/rawtransaction.cpp b/src/rpc/rawtransaction.cpp
index 5c93846..931eaeb 100644
--- a/src/rpc/rawtransaction.cpp
+++ b/src/rpc/rawtransaction.cpp
@@ -821 +821 @@ UniValue sendrawtransaction(const UniValue& params, bool fHelp)
-    CAmount txFee = 0;
+    CFeeRate txFee = CFeeRate(0);
diff --git a/src/wallet/wallet.cpp b/src/wallet/wallet.cpp
index 586fc69..0070ad1 100644
--- a/src/wallet/wallet.cpp
+++ b/src/wallet/wallet.cpp
@@ -1270 +1270 @@ bool CWalletTx::RelayWalletTransaction()
-            RelayTransaction((CTransaction)*this, feeRate.GetFeePerK());
+            RelayTransaction((CTransaction)*this, feeRate);